app/models/base_image.rb

```diff
@@ -39,6 +39,15 @@ def docker_index_url
     end
   end
 
+  def imagelayers_url
+    if remote?
```

Comment:
Not that big of a deal, but it would probably be better to put these implementations into the child classes that are inheriting from this base. The example set above (line 36) is a bad one, that creates a bi-directional dependency... This super class shouldn't really have knowledge of the child classes. There's also a missing test for the remote image case, I think. Though it gets tested implicitly in the as_json tests...